lib/parser/interface-list-parser.ts

```diff
@@ -126,9 +128,65 @@ export class InterfaceListParser extends ParserBase {
 
     }
     if (foundElements === 0) {
-      throw new ParserError(`Empty interface list is not allowed`, startToken.range.copyWithNewEnd(this.getToken().range));
+      let startOffset = 0;
+      while (this.getToken(startOffset, true) !== startToken) {
+        startOffset--;
+      }
+
+      if (this.getToken(startOffset - 1, true).getLText() === 'port'
+      || this.getToken(startOffset - 1, true).getLText() === 'generic'
+      || this.getToken(startOffset - 1, true).getLText() === 'parameter') {
+        startOffset--;
+      }
+      const closingBracket = this.getToken(-1, true);
+      let endDel = closingBracket;
+      if (this.getToken().getLText() === ';') {
+        endDel = this.getToken();
+      }
+      throw new ParserError(`Empty interface list is not allowed`,
+        startToken.range.copyWithNewEnd(closingBracket.range),
+        {
+          message: 'remove interface list',
+          edits: [
+            TextEdit.del(this.getToken(startOffset, true).range.copyWithNewEnd(endDel.range))
+          ]
+        }
+      );
     }
     this.debug('parseEnd');
 
   }
+  getTypeDefintion(parent: OGeneric | OPort) {
+    this.debug('getTypeDefintion');
+    const [type, last] = this.advanceParentheseAware([')', ';', ':='], true, false);
+    let defaultValue: OLexerToken[] = [];
+    if (last.getLText() === ':=') {
+      this.consumeToken();
+      [defaultValue] = this.advanceParentheseAware([')', ';'], true, false);
+    }
+    this.reverseWhitespace();
+    this.advanceWhitespace();
+    if (this.getToken().text === ';') {
+      const startI = this.pos.i;
+      this.consumeToken();
+      if (this.getToken().text === ')') {
+        const range = new OIRange(parent, startI, startI + 1).copyExtendBeginningOfLine();
+        throw new ParserError(`Unexpected ';' at end of interface list`, range, {
+          message: `Remove ';'`,
+          edits: [TextEdit.del(new OIRange(parent, startI, startI + 1))]
+        });
```

Comment:
when merging main (i.e. https://github.com/vhdl-linter/vhdl-linter/pull/165) this should be a diagnostic and not an error because it is recoverable.